test/IRGen/runtime_attributes_on_distributed_actors.swift

```diff
@@ -0,0 +1,47 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -emit-module-path=%t/RAD.swiftmodule -module-name=RAD -enable-experimental-feature RuntimeDiscoverableAttrs -disable-availability-checking %S/Inputs/runtime_attrs.swift
+// RUN: %target-swift-frontend -primary-file %s -emit-ir -I %t -swift-version 5 -enable-experimental-feature RuntimeDiscoverableAttrs -disable-availability-checking | %IRGenFileCheck %s
+
+// REQUIRES: asserts
+// REQUIRES: OS=macosx
+// REQUIRES: concurrency
+
+// CHECK: @"$s28runtime_attributes_on_actors9TestActorC15asyncExternallyyyKcvpfaAA17FlagForAsyncFuncsHF"
+// CHECK: @"$s28runtime_attributes_on_actors9TestActorC11doSomethingyyYaKcvpfaAA17FlagForAsyncFuncsHF"
+// CHECK: @"$s28runtime_attributes_on_actors9TestActorC11doSomethingyySiYacvpfaAA17FlagForAsyncFuncsHF"
+// CHECK: @"$s28runtime_attributes_on_actors9TestActorC11doSomething_1xySi_SaySiGztYacvpfaAA17FlagForAsyncFuncsHF"
+// CHECK: @"$s28runtime_attributes_on_actors13globalAsyncFnSaySSGyYacvpfaAA07FlagForF5FuncsHF"
+
+// CHECK: @"$s28runtime_attributes_on_actors17FlagForAsyncFuncsVHa" = internal constant
+// CHECK-SAME: i32 5
+// CHECK-SAME: %swift.accessible_function* @"$s28runtime_attributes_on_actors9TestActorC15asyncExternallyyyKcvpfaAA17FlagForAsyncFuncsHF"
+// CHECK-SAME: %swift.accessible_function* @"$s28runtime_attributes_on_actors9TestActorC11doSomethingyyYaKcvpfaAA17FlagForAsyncFuncsHF"
+// CHECK-SAME: %swift.accessible_function* @"$s28runtime_attributes_on_actors9TestActorC11doSomethingyySiYacvpfaAA17FlagForAsyncFuncsHF"
+// CHECK-SAME: %swift.accessible_function* @"$s28runtime_attributes_on_actors9TestActorC11doSomething_1xySi_SaySiGztYacvpfaAA17FlagForAsyncFuncsHF"
+// CHECK-SAME: %swift.accessible_function* @"$s28runtime_attributes_on_actors13globalAsyncFnSaySSGyYacvpfaAA07FlagForF5FuncsHF"
+
+import Distributed
+
```

Comment:
This test crashes:

```
+ /opt/homebrew/opt/python@3.10/bin/python3.10 /Users/ktoso/code/swift-project/swift/utils/PathSanitizingFileCheck --allow-unused-prefixes --sanitize BUILD_DIR=/Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64 --sanitize SOURCE_DIR=/Users/ktoso/code/swift-project/swift --use-filecheck /Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/FileCheck --check-prefix=CHECK --check-prefix=CHECK-macosx --check-prefix=CHECK-arm64 --check-prefix=CHECK-little --check-prefix=CHECK-64 --check-prefix=CHECK-noptrauth --check-prefix=CHECK-objc -DINT=i64 -D#CLASS_METADATA_HEADER=10 /Users/ktoso/code/swift-project/swift/test/IRGen/runtime_attributes_on_distributed_actors.swift
Global is external, but doesn't have external or weak linkage!
%swift.async_func_pointer* @"$s40runtime_attributes_on_distributed_actors9TestActorC15asyncExternallyyyKcvpfaAA17FlagForAsyncFuncsTFTu"
Global is external, but doesn't have external or weak linkage!
%swift.async_func_pointer* @"$s40runtime_attributes_on_distributed_actors9TestActorC11doSomethingyyYaKcvpfaAA17FlagForAsyncFuncsTFTu"
Global is external, but doesn't have external or weak linkage!
%swift.async_func_pointer* @"$s40runtime_attributes_on_distributed_actors9TestActorC11doSomething_1xySi_SaySiGtYacvpfaAA17FlagForAsyncFuncsTFTu"
<unknown>:0: error: fatal error encountered during compilation; please submit a bug report (https://swift.org/contributing/#reporting-bugs)
<unknown>:0: note: Broken module found, compilation aborted!
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.	Program arguments: /Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/bin/swift-frontend -target arm64-apple-macosx10.9 -module-cache-path /Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/swift-test-results/arm64-apple-macosx10.9/clang-module-cache -sdk /Applications/Xcode-14-Seed-Latest-e.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk -swift-version 4 -define-availability "SwiftStdlib 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999" -define-availability "SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2" -define-availability "SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0" -define-availability "SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4" -define-availability "SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0" -define-availability "SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5" -define-availability "SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0" -define-availability "SwiftStdlib 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4" -define-availability "SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0" -define-availability "SwiftStdlib 5.8:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999" -define-availability "SwiftStdlib 5.9:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999" -typo-correction-limit 10 -primary-file /Users/ktoso/code/swift-project/swift/test/IRGen/runtime_attributes_on_distributed_actors.swift -emit-ir -I /Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64/IRGen/Output/runtime_attributes_on_distributed_actors.swift.tmp -swift-version 5 -enable-experimental-feature RuntimeDiscoverableAttrs -disable-availability-checking
1.	Swift version 5.9-dev (LLVM dac17a9b87acc8e, Swift 4092a99aa372fed)
2.	Compiling with the current language version
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x0000000104fbcd6c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x0000000104fbbf60 llvm::sys::RunSignalHandlers() + 112
2  swift-frontend           0x0000000104fbd3ac SignalHandler(int) + 304
3  libsystem_platform.dylib 0x0000000192986a84 _sigtramp + 56
4  libsystem_pthread.dylib  0x0000000192957c28 pthread_kill + 288
5  libsystem_c.dylib        0x0000000192865b18 abort + 180
6  swift-frontend           0x00000001003439cc PrettyStackTraceFrontend::~PrettyStackTraceFrontend() + 0
7  swift-frontend           0x0000000104f06b18 llvm::report_fatal_error(llvm::Twine const&, bool) + 268
8  swift-frontend           0x0000000104f06a0c llvm::report_fatal_error(llvm::Twine const&, bool) + 0
9  swift-frontend           0x0000000104d00dd0 llvm::VerifierPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) + 0
10 swift-frontend           0x0000000104cdba9c llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) + 396
11 swift-frontend           0x000000010075ccb0 swift::performLLVMOptimizations(swift::IRGenOptions const&, llvm::Module*, llvm::TargetMachine*) + 4544
12 swift-frontend           0x000000010075de80 swift::performLLVM(swift::IRGenOptions const&, swift::DiagnosticEngine&, llvm::sys::SmartMutex<false>*, llvm::GlobalVariable*, llvm::Module*, llvm::TargetMachine*, llvm::StringRef, swift::UnifiedStatsReporter*) + 2764
13 swift-frontend           0x000000010033db50 performCompileStepsPostSILGen(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule> >, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*) + 2460
14 swift-frontend           0x000000010033cc10 swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 564
15 swift-frontend           0x000000010034b940 withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) + 160
16 swift-frontend           0x000000010033e78c swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 2888
17 swift-frontend           0x000000010030d784 swift::mainEntry(int, char const**) + 3456
18 dyld                     0x00000001925fff28 start + 2236
FileCheck error: '<stdin>' is empty.
FileCheck command line:  /Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/bin/FileCheck --allow-unused-prefixes --check-prefix=CHECK --check-prefix=CHECK-macosx --check-prefix=CHECK-arm64 --check-prefix=CHECK-little --check-prefix=CHECK-64 --check-prefix=CHECK-noptrauth --check-prefix=CHECK-objc -DINT=i64 -D#CLASS_METADATA_HEADER=10 /Users/ktoso/code/swift-project/swift/test/IRGen/runtime_attributes_on_distributed_actors.swift
```

Reply:
I think this is actually a problem with distributed actors and not attributes...